### ISSUE TITLE
Saving forces login even if not needed by backend

### ISSUE THREAD

**Issue by valterkraemer** [mavoweb/mavo#245]
I'm working on a backend plugin. There are scenarios when you could, but don't need to log in to store data. Now Mavo forces me to always log in if it's possible to.

https://github.com/mavoweb/mavo/blob/master/src/mavo.js#L488

**Comment by joyously** [comment]
Shouldn't your backend plugin just override that function?

**Comment by valterkraemer** [comment]
The problem is in `mavo.js`, not in `backend.js`, correct me if I'm wrong but I don't think I should override it, and in that case, how?

**Comment by joyously** [comment]
The github backend provides a function for login and getUser.

**Comment by valterkraemer** [comment]
And how does that help me? Mavo.js calls the same login function in backend.js (that I can override) both on store and when login-button is pressed.

In my backend the user doesn't need to be logged in (but have the option to) to be able to store data.

**Comment by LeaVerou** [comment]
@valterkraemer is correct. This is a bug (well, more like shortsightedness on my part when I wrote that bit). I was expecting that as we see more backends rolling in, all the false assumptions we had made in the code would be uncovered one by one :)

@valterkraemer since you've been sending PRs (thanks!), I think this can be easily fixed by just removing `.login()` there. There are very few cases where this matters, namely:

- Backends that enable editing and disable saving for logged out users
- Apps that contain form elements (which are always editable, intrinsically) and the user tries to save by hitting Cmd+S
